File: greatest-common-divisor.py
#!/usr/bin/env python3
"""
Greatest Common Divisor Calculator (GCD)

This script computes the Greatest Common Divisor (GCD) of two or more integers
using the Euclidean algorithm. It supports input via the command line with
space-separated or comma-separated values.

Features:
- Default input: [48, 18, 30] if none provided.
- Validates input and handles non-integer values gracefully.
- Uses custom implementation of Euclidean algorithm.

Author: Scott
Date: 2025-07-20
"""

def gcd(a, b):
    """Compute the greatest common divisor using the Euclidean algorithm."""
    while b != 0:
        a, b = b, a % b
    return abs(a)

def gcd_list(numbers):
    """Compute GCD for a list of integers."""
    if not numbers:
        return 0
    result = numbers[0]
    for num in numbers[1:]:
        result = gcd(result, num)
    return result

def parse_input(input_str):
    """Convert comma or space-separated string into a list of integers."""
    try:
        cleaned = input_str.replace(',', ' ')
        return [int(x) for x in cleaned.split() if x]
    except ValueError as e:
        print(f"❌ Invalid input: {e}")
        exit(1)

def main():
    input_str = input("Enter integers (comma or space separated, default: 48 18 30): ").strip()
    numbers = parse_input(input_str) if input_str else [48, 18, 30]
    result = gcd_list(numbers)
    joined = ', '.join(str(n) for n in numbers)
    print(f"GCD({joined}) = {result}")

if __name__ == "__main__":
    main()

File: greatest-common-divisor.cpp
/*
 * Greatest Common Divisor Calculator (GCD)
 *
 * This program computes the Greatest Common Divisor (GCD) of two or more integers
 * using the Euclidean algorithm. It accepts input via standard input, with
 * support for both space-separated and comma-separated integers.
 *
 * Author: Scott
 * Date: 2025-07-20
 */

#include <iostream>
#include <sstream>
#include <vector>
#include <string>
#include <algorithm>
#include <cstdlib>
#include <cctype>

int gcd(int a, int b) {
    while (b != 0) {
        int tmp = b;
        b = a % b;
        a = tmp;
    }
    return std::abs(a);
}

int gcd_list(const std::vector<int>& nums) {
    if (nums.empty()) return 0;
    int result = nums[0];
    for (size_t i = 1; i < nums.size(); ++i) {
        result = gcd(result, nums[i]);
    }
    return result;
}

std::vector<int> parse_input(const std::string& input) {
    std::vector<int> numbers;
    std::string cleaned;
    for (char ch : input) {
        if (ch == ',') cleaned += ' ';
        else cleaned += ch;
    }

    std::istringstream iss(cleaned);
    std::string token;
    while (iss >> token) {
        try {
            numbers.push_back(std::stoi(token));
        } catch (...) {
            std::cerr << "❌ Invalid input: '" << token << "' is not an integer.\n";
            exit(1);
        }
    }
    return numbers;
}

int main() {
    std::string input;
    std::cout << "Enter integers (space or comma separated, default: 48 18 30): ";
    std::getline(std::cin, input);

    std::vector<int> numbers = input.empty() ? std::vector<int>{48, 18, 30} : parse_input(input);
    int result = gcd_list(numbers);

    std::cout << "GCD(";
    for (size_t i = 0; i < numbers.size(); ++i) {
        std::cout << numbers[i];
        if (i != numbers.size() - 1) std::cout << ", ";
    }
    std::cout << ") = " << result << std::endl;

    return 0;
}
